Operator 0.13.0 crashes with go panic

<!-- Please answer these questions before submitting your bug report. Thanks! -->

**What version of redis operator are you using?**

```shell
kubectl logs <_redis-operator_pod_name> -n <namespace>

{"level":"info","ts":1669194760.4806912,"logger":"controller-runtime.metrics","msg":"Metrics server is starting to listen","addr":":8080"}
{"level":"info","ts":1669194760.4818819,"logger":"setup","msg":"starting manager"}
{"level":"info","ts":1669194760.4829562,"msg":"Starting server","path":"/metrics","kind":"metrics","addr":"[::]:8080"}
{"level":"info","ts":1669194760.4829874,"msg":"Starting server","kind":"health probe","addr":"[::]:8081"}
I1123 09:12:40.483057       1 leaderelection.go:248] attempting to acquire leader lease redis-operator/6cab913b.redis.opstreelabs.in...
I1123 09:12:58.319509       1 leaderelection.go:258] successfully acquired lease redis-operator/6cab913b.redis.opstreelabs.in
{"level":"info","ts":1669194778.319815,"logger":"controller.redis","msg":"Starting EventSource","reconciler group":"redis.redis.opstreelabs.in","reconciler kind":"Redis","source":"kind source: *v1beta1.Redis"}
{"level":"info","ts":1669194778.3198977,"logger":"controller.redis","msg":"Starting Controller","reconciler group":"redis.redis.opstreelabs.in","reconciler kind":"Redis"}
{"level":"info","ts":1669194778.3198225,"logger":"controller.rediscluster","msg":"Starting EventSource","reconciler group":"redis.redis.opstreelabs.in","reconciler kind":"RedisCluster","source":"kind source: *v1beta1.RedisCluster"}
{"level":"info","ts":1669194778.319965,"logger":"controller.rediscluster","msg":"Starting Controller","reconciler group":"redis.redis.opstreelabs.in","reconciler kind":"RedisCluster"}
{"level":"info","ts":1669194778.4202948,"logger":"controller.redis","msg":"Starting workers","reconciler group":"redis.redis.opstreelabs.in","reconciler kind":"Redis","worker count":1}
{"level":"info","ts":1669194778.4217591,"logger":"controller.rediscluster","msg":"Starting workers","reconciler group":"redis.redis.opstreelabs.in","reconciler kind":"RedisCluster","worker count":1}
{"level":"info","ts":1669194778.421855,"logger":"controllers.RedisCluster","msg":"Reconciling opstree redis Cluster controller","Request.Namespace":"default","Request.Name":"redis-cluster"}
{"level":"info","ts":1669194778.4259007,"logger":"controller_redis","msg":"Redis statefulset get action was successful","Request.StatefulSet.Namespace":"default","Request.StatefulSet.Name":"redis-cluster-leader"}
{"level":"info","ts":1669194778.4380147,"logger":"controller_redis","msg":"Reconciliation Complete, no Changes required.","Request.StatefulSet.Namespace":"default","Request.StatefulSet.Name":"redis-cluster-leader"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x140bd62]

goroutine 202 [running]:
redis-operator/k8sutils.RedisClusterService.CreateRedisClusterService({{0x172857b, 0x19726d8}}, 0xc000154f00)
	/workspace/k8sutils/redis-cluster.go:180 +0x3a2
redis-operator/k8sutils.CreateRedisLeaderService(...)
	/workspace/k8sutils/redis-cluster.go:132
redis-operator/controllers.(*RedisClusterReconciler).Reconcile(0xc000884090, {0xc000695740, 0x155f2e0}, {{{0xc000613d16, 0x166a200}, {0xc000613d20, 0x30}}})
	/workspace/controllers/rediscluster_controller.go:78 +0x370
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0xc0005a28f0, {0x1944818, 0xc000695740}, {{{0xc000613d16, 0x166a200}, {0xc000613d20, 0x413a34}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:114 +0x26f
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0005a28f0, {0x1944770, 0xc000891ec0}, {0x15b4b40, 0xc00071e3c0})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:311 +0x33e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0005a28f0, {0x1944770, 0xc000891ec0})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:266 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:227 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.11.0/pkg/internal/controller/controller.go:223 +0x357
```

redis-operator version: 0.13.0

**Does this issue reproduce with the latest release?**
yes

**What operating system and processor architecture are you using (`kubectl version`)?**

<details><summary><code>kubectl version</code>Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.2", GitCommit:"f66044f4361b9f1f96f0053dd46cb7dce5e990a8", GitTreeState:"clean", BuildDate:"2022-06-15T14:22:29Z", GoVersion:"go1.18.3", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.13", GitCommit:"592eca05be27f7d927d0b25cbb4241d75a9574bf", GitTreeState:"clean", BuildDate:"2022-10-26T15:19:38Z", GoVersion:"go1.17.13", Compiler:"gc", Platform:"linux/amd64"}
</summary><br><pre>
$ kubectl version

</pre></details>

**What did you do?**

I applied the following cluster

```
apiVersion: redis.redis.opstreelabs.in/v1beta1
kind: RedisCluster
metadata:
  name: redis-cluster
spec:
  clusterSize: 3
  persistenceEnabled: true
  clusterVersion: v7
  redisLeader:
    replicas: 3
    pdb:
      enabled: true
      maxUnavailable: 1
  redisFollower:
    replicas: 3
    pdb:
      enabled: true
      maxUnavailable: 1
  redisExporter:
    enabled: false
    image: "quay.io/opstree/redis-exporter:v1.44.0"
    imagePullPolicy: "IfNotPresent"
    resources:
      {}
  kubernetesConfig:
    image: "quay.io/opstree/redis:v7.0.5"
    imagePullPolicy: "IfNotPresent"
    resources:
      {}
  storage:
    volumeClaimTemplate:
      spec:
        accessModes:
        - ReadWriteOnce
        resources:
          requests:
            storage: 1Gi
        storageClassName: sinapse-standard
  securityContext:
    fsGroup: 1000
    runAsUser: 1000
```



**What did you expect to see?**

a running redis cluster after some time

**What did you see instead?**
Operator crashing

One hint, with 0.12.0 it works.

--- a/k8sutils/redis-cluster.go
+++ b/k8sutils/redis-cluster.go
@@ -175,9 +175,13 @@ func (service RedisClusterService) CreateRedisClusterService(cr *redisv1beta1.Re
 	if cr.Spec.RedisExporter != nil && cr.Spec.RedisExporter.Enabled {
 		enableMetrics = true
 	}
+	additionalServiceAnnotations := map[string]string{}
+	if cr.Spec.KubernetesConfig.Service != nil {
+		additionalServiceAnnotations = cr.Spec.KubernetesConfig.Service.ServiceAnnotations
+	}
 	objectMetaInfo := generateObjectMetaInformation(serviceName, cr.Namespace, labels, annotations)
 	headlessObjectMetaInfo := generateObjectMetaInformation(serviceName+"-headless", cr.Namespace, labels, annotations)
-	additionalObjectMetaInfo := generateObjectMetaInformation(serviceName+"-additional", cr.Namespace, labels, generateServiceAnots(cr.ObjectMeta, cr.Spec.KubernetesConfig.Service.ServiceAnnotations))
+	additionalObjectMetaInfo := generateObjectMetaInformation(serviceName+"-additional", cr.Namespace, labels, generateServiceAnots(cr.ObjectMeta, additionalServiceAnnotations))
 	err := CreateOrUpdateService(cr.Namespace, headlessObjectMetaInfo, redisClusterAsOwner(cr), false, true, "ClusterIP")
 	if err != nil {
 		logger.Error(err, "Cannot create headless service for Redis", "Setup.Type", service.RedisServiceRole)

--- a/k8sutils/redis-cluster.go
+++ b/k8sutils/redis-cluster.go
@@ -192,7 +192,11 @@ func (service RedisClusterService) CreateRedisClusterService(cr *redisv1beta1.Re
 		logger.Error(err, "Cannot create service for Redis", "Setup.Type", service.RedisServiceRole)
 		return err
 	}
-	err = CreateOrUpdateService(cr.Namespace, additionalObjectMetaInfo, redisClusterAsOwner(cr), false, false, cr.Spec.KubernetesConfig.Service.ServiceType)
+	additionalServiceType := "ClusterIP"
+	if cr.Spec.KubernetesConfig.Service != nil {
+		additionalServiceType = cr.Spec.KubernetesConfig.Service.ServiceType
+	}
+	err = CreateOrUpdateService(cr.Namespace, additionalObjectMetaInfo, redisClusterAsOwner(cr), false, false, additionalServiceType)
 	if err != nil {
 		logger.Error(err, "Cannot create additional service for Redis", "Setup.Type", service.RedisServiceRole)
 		return err

--- a/k8sutils/redis-standalone.go
+++ b/k8sutils/redis-standalone.go
@@ -16,9 +16,13 @@ func CreateStandaloneService(cr *redisv1beta1.Redis) error {
 	if cr.Spec.RedisExporter != nil && cr.Spec.RedisExporter.Enabled {
 		enableMetrics = true
 	}
+	additionalServiceAnnotations := map[string]string{}
+	if cr.Spec.KubernetesConfig.Service != nil {
+		additionalServiceAnnotations = cr.Spec.KubernetesConfig.Service.ServiceAnnotations
+	}
 	objectMetaInfo := generateObjectMetaInformation(cr.ObjectMeta.Name, cr.Namespace, labels, annotations)
 	headlessObjectMetaInfo := generateObjectMetaInformation(cr.ObjectMeta.Name+"-headless", cr.Namespace, labels, annotations)
-	additionalObjectMetaInfo := generateObjectMetaInformation(cr.ObjectMeta.Name+"-additional", cr.Namespace, labels, generateServiceAnots(cr.ObjectMeta, cr.Spec.KubernetesConfig.Service.ServiceAnnotations))
+	additionalObjectMetaInfo := generateObjectMetaInformation(cr.ObjectMeta.Name+"-additional", cr.Namespace, labels, generateServiceAnots(cr.ObjectMeta, additionalServiceAnnotations))
 	err := CreateOrUpdateService(cr.Namespace, headlessObjectMetaInfo, redisAsOwner(cr), false, true, "ClusterIP")
 	if err != nil {
 		logger.Error(err, "Cannot create standalone headless service for Redis")

--- a/k8sutils/redis-standalone.go
+++ b/k8sutils/redis-standalone.go
@@ -33,7 +33,11 @@ func CreateStandaloneService(cr *redisv1beta1.Redis) error {
 		logger.Error(err, "Cannot create standalone service for Redis")
 		return err
 	}
-	err = CreateOrUpdateService(cr.Namespace, additionalObjectMetaInfo, redisAsOwner(cr), false, false, cr.Spec.KubernetesConfig.Service.ServiceType)
+	additionalServiceType := "ClusterIP"
+	if cr.Spec.KubernetesConfig.Service != nil {
+		additionalServiceType = cr.Spec.KubernetesConfig.Service.ServiceType
+	}
+	err = CreateOrUpdateService(cr.Namespace, additionalObjectMetaInfo, redisAsOwner(cr), false, false, additionalServiceType)
 	if err != nil {
 		logger.Error(err, "Cannot create additional service for Redis")
 		return err

